### PR TITLE
@damassi => Redux images

### PR DESCRIPTION
**Pull request by eessex**
Converts Images components to use Redux data and actions. Also makes a small change to drag/drop component for better unique keys.

### CODE REVIEW

File: client/apps/edit2/components/content/sections/images/components/controls.jsx [hunk 1]
```diff
@@ -63,10 +78,21 @@ export class ImagesControls extends Component {
...
+    } = this.props
 
-    section.set('images', newImages)
+    if (isHero) {
+      const newImages = clone(section.images).concat(image)
+      onChangeHeroAction('images', newImages)
+    } else {
+      const newImages = clone(editSection.images).concat(image)
+      onChangeSectionAction('images', newImages)
+    }
```
Comment (damassi) [inline client/apps/edit2/components/content/sections/images/components/controls.jsx]: 👍this is the sort of thing that could have been written cleverly but straightforward and clear is always better

File: client/apps/edit2/components/content/sections/images/components/controls.jsx [hunk 2]
```diff
@@ -93,43 +120,43 @@ export class ImagesControls extends Component {
...
-          section={section}
-          articleLayout={articleLayout}
-          sectionLayouts={!isHero}
+          showLayouts={!isHero}
           isHero={isHero}
-          disabledAlert={this.fillwidthAlert}
+          disabledAlert={this.fillWidthAlert}
         >
-          <div onClick={inputsAreDisabled ? this.fillwidthAlert : undefined}>
+          <div onClick={inputsAreDisabled ? this.fillWidthAlert : undefined}>
```
Comment (damassi) [inline client/apps/edit2/components/content/sections/images/components/controls.jsx]: I wonder if `{inputsAreDisabled && this.fillWidthAlert}` would work

Comment (eessex) [reply client/apps/edit2/components/content/sections/images/components/controls.jsx]: I get a console error with that, for functions it wants undefined to be explicitly declared.

File: client/apps/edit2/components/content/sections/images/test/edit_image.test.js [hunk 3]
```diff
@@ -22,80 +22,104 @@ describe('EditImage', () => {
...
+    })
+
+    it('renders the remove button if editing and props.removeItem', () => {
+      props.editing = true
+      const component = getWrapper(props)
+
+      expect(component.find(RemoveButton).exists()).toBe(true)
+    })
+
+    xit('calls removeItem when clicking remove icon', () => {
```
Comment (damassi) [inline client/apps/edit2/components/content/sections/images/test/edit_image.test.js]: Deliberate?

Comment (eessex) [reply client/apps/edit2/components/content/sections/images/test/edit_image.test.js]: For some reason this test started failing when I broke the code out from the main PR -- figured I would come back to it when I merge edit with edit2 and everything's in place.

Comment (damassi) [reply client/apps/edit2/components/content/sections/images/test/edit_image.test.js]: cool cool

**Pull request merged**